e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/CachedConstraintRun.java

```diff
@@ -18,7 +18,7 @@ public static CachedConstraintRun fromJSON(JsonObject json) {
         json.getBoolean("definition_outdated"),
         json.getJsonObject("results").isEmpty() ?
             Optional.empty() :
-            Optional.of(ConstraintRecord.fromJSON(json.getJsonObject("results")))
+            Optional.of(ConstraintResult.fromJSON(json.getJsonObject("results")))
```

Comment:
Just to confirm: the reason why the Cached Results no longer equal what is returned from the endpoint is because we only want to cache successful constraint runs (as in, there were no errors during execution)? If so, will #1194 cause any complications?

Reply:
Since we weren't storing failed runs in the first place, I don't see this causing any issues. The original behavior would impose constraints that only compiled correctly, preventing compilation errors from being stored in the cache or database. We only care about runs that compiled and executed successfully on the server. With #1194, we'll filter out any errors and only store successful runs to the database and cache..